.github/workflows/push.yml

```diff
@@ -25,15 +47,47 @@ jobs:
       - name: Build project
         run: ~/.elan/bin/lake -Kenv=dev build LeanAPAP
 
+      - uses: actions/cache@v3
+        name: Mathlib doc Cache
+        with:
+          path: build/doc/Mathlib
+          key: DocGen4-${{ hashFiles('lake-manifest.json') }}
+          restore-keys: |
+            DocGen4-
+
       - name: Build documentation
         run: ~/.elan/bin/lake -Kenv=dev build LeanAPAP:docs
 
-      - name: Copy documentation
-        run: cp -R build/doc docs/
+      - name: Install Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.9'
+          cache: 'pip' # caching pip dependencies
+
+      - name: Install blueprint apt deps
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y graphviz libgraphviz-dev pdf2svg dvisvgm texlive-full
+
+      - name: Install blueprint deps
+        run: |
+          cd blueprint && pip install -r requirements.txt
+
+      - name: Build blueprint and copy to `docs/blueprint`
+        run: |
+          inv all
```

Comment:
I noticed your `all` tasks would copy blueprint to `docs/blueprint`, so it's also moved by the next step.